### PR TITLE
BCDA-4323 - Add service config struct in service package

### PR DESCRIPTION
**Pull request by mtrang1263**
### Part of [BCDA-4323](https://jira.cms.gov/browse/BCDA-4323)

We need to add a configuration struct to our service to capture the lookback window. To do this, we introduced the config struct that holds the ACO based lookback window and a set of configuration variables (cutoff date, suppression lookback, runout cutoff date, and runout service).

### Change Details
1. Update our conf package to merge multiple config file sources.
2. Replaced our custom struct parsing to leverage the unmarshalling capabilities provided by viepr.
3. Add config.go to our service package to capture the fields described above.

### Security Implications

- [x] new software dependencies
Added github.com/mitchellh/mapstructure which is used to unmarshal a configuration struct using Viper.

- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
1. Will deploy to dev and verify that we can load our remote configuration as expected. Will verify the behavior of this [PR](https://github.com/CMSgov/bcda-ops/pull/680).
2. Run smoke tests on dev environment to verify configs are parsed correctly.

### CODE REVIEW

File: conf/config.go [hunk 1]
```diff
@@ -277,38 +288,41 @@ func Checkout(v interface{}) error {
...
-			if name == "-" {
-				continue
-			} else if name == "" {
-				// If the Tag is not there, then default to the name of the struct
-				name = fieldType.Field(i).Name
+// bindenv: workaround to make the unmarshal work with environment variables
+// Inspired from solution found here : https://github.com/spf13/viper/issues/188#issuecomment-399884438
+func bindenvs(field reflect.Value, parts ...string) error {
+	if field.Kind() == reflect.Ptr {
+		field = field.Elem()
```
Comment (CosmicSyntax) [inline conf/config.go:295]: Do you have any, or is there any chance, we will have a config variable behind a pointer / reference?  If not, I wonder if we can just replace this with a continue and skip that field?

**Review by CosmicSyntax — APPROVED** [review]
LGTM!

**Review by katyasoup — APPROVED** [review]
LGTM!

**Review by SMLuthi — APPROVED** [review]
This looks great!

**Pull request merged**